src/protocols/BaseProtocolAdapter.ts

```diff
@@ -96,6 +97,19 @@ export abstract class BaseProtocolAdapter {
 
   abstract fetchAccountTokenBalances(address: string): Promise<ITokenBalance[]>;
 
+  abstract transferToken(
+    amount: number,
+    recipient: string,
+    contractId: string,
+    options: {
+      fromAccount: IAccount;
```

Comment:
Maybe instead of `IAccount` it would be better to use `string` and pass the address?

Reply:
Are you suggesting to get the `secretKey` in the function itself using the 'address'?

Reply:
This is what we are doing here: https://github.com/superhero-com/superhero-wallet/blob/develop/src/protocols/aeternity/libs/AeAccountHdWallet.ts#L21
I think we need to be consistent on the type passed here. If you think passing whole account is more reasonable I think the `AeAccountHdWallet` needs to be updated also.

Reply:
Same approach is chosen in `spend` method here. https://github.com/superhero-com/superhero-wallet/blob/feat/eth-support/src/protocols/ethereum/libs/EthereumAdapter.ts#L233
in order to get the `secretKey` like that
https://github.com/superhero-com/superhero-wallet/blob/feat/eth-support/src/protocols/ethereum/libs/EthereumAdapter.ts#L262
Do you want to reconsider it and for me to apply some changes on both of them?

Reply:
Whatever we choose I think we need to be consistent. In my opinion passing the address and then picking the account based on the address is more safe.